app/models/bib.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+# Model for bib-level information in Symphony, it's basically a hold request factory
+class Bib
+  include BibRecord
+  attr_reader :body, :barcode
+
+  def initialize(body, holdable_locations)
+    @body = body
+    @barcode = random_barcode
+    @holdable_locations = holdable_locations
+  end
+
+  # Take the array of holds and find out what is holdable. Order is important.
+  #
+  #  * Must be more than 1 overall potential holdables.
+  #  * Each potential holdable must have a current location that is holdable.
+  #  * To be a holdable set:
+  #    * Must be at least one volumetric holdable.
+  #    * Must be more than one holdables.
+  #
+  # In the case where a holdable set is not discovered, nil is returned and the barcode instance attribute
+  # is available to process the hold request.
+  def holdables
+    return nil unless potential_holdables.count > 1
+
+    filter_holdables
+
+    return nil unless volumetric? && @holds.count > 1
+
+    @holds # .sort_by {|a, b| a.key <=> b.key }
```

Comment:
@cdmo sort by will in another branch right?

Reply:
yes, remove this commented out sort, sorry